python/plugins/processing/algs/qgis/PointOnSurface.py

```diff
@@ -66,26 +67,24 @@ def displayName(self):
         return self.tr('Point on surface')
 
     def processAlgorithm(self, parameters, context, feedback):
-        layer = QgsProcessingUtils.mapLayerFromString(self.getParameterValue(self.INPUT_LAYER), context)
+        source = self.parameterAsSource(self.INPUT_LAYER, context)
```

Comment:
this should be

    source = self.parameterAsSource(parameters, self.INPUT_LAYER, context)